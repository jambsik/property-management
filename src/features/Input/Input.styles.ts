import { Theme } from '@mui/system';

export const inputRootStyles = (theme: Theme) => ({
    input: {
        '&:hover': {
            color: theme.palette.info.main,
        },
    },
});
