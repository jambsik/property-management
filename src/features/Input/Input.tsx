import { TextField, TextFieldProps } from '@mui/material';
import React from 'react';
import { inputRootStyles } from './Input.styles';

export interface InputProps {
    name: string;
    label: string;
    type: string;
}

const Input = ({ name, label, type, ...props }: InputProps | TextFieldProps) => {
    return <TextField sx={inputRootStyles} id={name} label={label} type={type} variant="standard" color="info" {...props} />;
};

export default Input;
