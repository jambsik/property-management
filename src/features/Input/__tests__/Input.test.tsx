import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Input from '../Input';
describe('Imput component test', () => {
    const inputProps = {
        label: 'My Input Test',
        name: 'input-test',
        type: 'number',
    };

    const pageObject = {
        input: 'input[id="input-test"]',
    };

    it('Should render component without errors', () => {
        const { container } = render(<Input {...inputProps} />);
        expect(container).toMatchSnapshot();
    });

    it('Should render label text', () => {
        const { container } = render(<Input {...inputProps} />);

        expect(container).toHaveTextContent(inputProps.label);
    });

    it('Should update input value', () => {
        const { container } = render(<Input {...inputProps} />);
        const inputElement = container.querySelector(pageObject.input);

        expect(inputElement).toBeInTheDocument();

        fireEvent.change(inputElement!, {
            target: {
                value: 24,
            },
        });

        expect(inputElement).toHaveValue(24);
    });

    it('Should validate if the type is numeric it should not be affected by the change', () => {
        const { container } = render(<Input {...inputProps} />);
        const inputElement = container.querySelector(pageObject.input);

        expect(inputElement).toBeInTheDocument();

        fireEvent.change(inputElement!, {
            target: {
                value: 'test',
            },
        });

        expect(inputElement).toHaveValue(null);
    });
});
