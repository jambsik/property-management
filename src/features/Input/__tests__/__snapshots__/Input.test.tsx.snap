// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Imput component test Should render component without errors 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-1954myt-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorInfo css-ahcq22-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="input-test"
      id="input-test-label"
    >
      My Input Test
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorInfo MuiInputBase-formControl css-ovthwe-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="input-test"
        type="number"
        value=""
      />
    </div>
  </div>
</div>
`;
